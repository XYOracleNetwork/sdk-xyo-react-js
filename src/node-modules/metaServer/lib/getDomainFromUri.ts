import { URL } from 'url'
export const getDomainFromUri = (uri: string) => {
  const parsed = new URL(uri)
  return `${parsed.protocol}//${parsed.host}`
}
