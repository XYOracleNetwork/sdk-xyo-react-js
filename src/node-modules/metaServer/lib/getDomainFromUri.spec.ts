import { getDomainFromUri } from './getDomainFromUri'

const validUris = [
  ['https://www.google.com/search/foo', 'https://www.google.com'],
  ['https://www.google.com', 'https://www.google.com'],
  [
    'https://beta.explore.xyo.network/archive/temp/block/hash/18b79aa80de2bcc7287fd0577a5c2caee47a37fd4ac75a252e4bc05aeddd7fbb?network=kerplunk',
    'https://beta.explore.xyo.network',
  ],
  ['https://explore.xyo.network/archive/temp/block/hash/ce9d1723d3f501abc980c4f48248c11c9689051d9a2010d521e4ee484b260436?network=main', 'https://explore.xyo.network'],
  ['https://api.archivist.xyo.network', 'https://api.archivist.xyo.network'],
]

describe('getDomainFromUri', () => {
  it.each(validUris)('gets domain from URI', (uri, expected) => {
    expect(getDomainFromUri(uri)).toBe(expected)
  })
})
