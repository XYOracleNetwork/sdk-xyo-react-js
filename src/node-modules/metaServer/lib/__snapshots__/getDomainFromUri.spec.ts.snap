// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDomainFromUri gets domain from URI 1`] = `"https://www.google.com"`;

exports[`getDomainFromUri gets domain from URI 2`] = `"https://www.google.com"`;

exports[`getDomainFromUri gets domain from URI 3`] = `"https://beta.explore.xyo.network"`;

exports[`getDomainFromUri gets domain from URI 4`] = `"https://explore.xyo.network"`;

exports[`getDomainFromUri gets domain from URI 5`] = `"https://api.archivist.xyo.network"`;
